app/tests/components_tests/test_views.py

```diff
@@ -190,3 +191,75 @@ def test_interfaces_create_permissions(
         user=editor,
     )
     assert response.status_code == 200
+
+
+@pytest.mark.parametrize(
+    "base_object_factory,base_obj_lookup,object_factory,viewname",
+    (
+        (
+            ReaderStudyFactory,
+            "reader_study",
+            DisplaySetFactory,
+            "reader-studies:display-set-delete",
+        ),
+        (
+            ArchiveFactory,
+            "archive",
+            ArchiveItemFactory,
+            "archives:item-delete",
+        ),
+    ),
+)
+@pytest.mark.django_db
+def test_civset_delete_view(
+    client, base_object_factory, base_obj_lookup, object_factory, viewname
+):
+    user, editor, collaborator = UserFactory.create_batch(3)
+    base_obj = base_object_factory()
+    obj = object_factory(**{base_obj_lookup: base_obj})
+    base_obj.add_editor(editor)
+    if base_object_factory == ReaderStudyFactory:
+        base_obj.add_reader(collaborator)
+    elif base_object_factory == ArchiveFactory:
+        base_obj.add_uploader(collaborator)
+
+    response = get_view_for_user(
+        viewname=viewname,
+        client=client,
+        user=user,
+        reverse_kwargs={"slug": base_obj.slug, "pk": obj.pk},
+    )
+    assert response.status_code == 403
+
+    response = get_view_for_user(
+        viewname=viewname,
+        client=client,
+        user=collaborator,
+        reverse_kwargs={"slug": base_obj.slug, "pk": obj.pk},
+    )
+    if base_object_factory == ReaderStudyFactory:
+        assert response.status_code == 403
+    elif base_object_factory == ArchiveFactory:
+        assert response.status_code == 200
```

Comment:
Why do uploaders get to delete? I would have thought they would only be able to upload.

Reply:
I wasn't sure about this. My reasoning was the whoever can create and edit archive items should also be able to delete them. Both editors and uploaders can create and edit. But I'm also fine with restricting the delete view to editors.

Reply:
I always saw uploaders as system users who were pushing data from external systems. I think that it is fine not to give them delete permission.